cli/cppcheckexecutor.cpp

```diff
@@ -71,6 +72,117 @@
 #endif
 
 namespace {
+    class SarifReport {
+    public:
+        void addFinding(ErrorMessage msg) {
+            mFindings.push_back(std::move(msg));
+        }
+
+        picojson::array serializeRules() const {
+            picojson::array ret;
+            std::set<std::string> ruleIds;
+            for (const auto& finding : mFindings) {
+                if (ruleIds.insert(finding.id).second) {
+                    picojson::object rule;
+                    rule["id"] = picojson::value(finding.id);
+                    picojson::object shortDescription;
+                    shortDescription["text"] = picojson::value(finding.shortMessage());
+                    rule["shortDescription"] = picojson::value(shortDescription);
+                    ret.emplace_back(rule);
+                }
+            }
+            return ret;
+        }
```

Comment:
`serializeRules()` does not serialize the rule's `fullDescription.text` nor its `help.text`, which are both required\*. See [`reportingDescriptor` object specification](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#reportingdescriptor-object).

Also, while not required, it would be nice to see the severity (`properties.problem.severity`) implemented as well.


\* These properties are required by GitHub's Code scanning platform, not necessarily the SARIF specification.

Reply:
ok